.github/workflows/tests.yml

```diff
@@ -51,13 +51,17 @@ jobs:
           rm -rf dist/*
           python setup.py sdist
           pip install dist/*.tar.gz
+          pushd ..
           python -c "import conda_smithy; assert conda_smithy.__version__ != '0.0.0'"
```

Comment:
https://twitter.com/ocefpaf/status/1562400506437373953

Reply:
bahahahahahahahahahah

I am so confused right now. What is all of this stuff for? Why does the sdist have 0.0.0 in the name? Where do I put anything?

Reply:
The sdist with build has build isolation, so no setuptools_scm is installed and no version is recorded in the name. The direct call to setup.py uses whatever you have install so, if you have setuptools_scm you will get a version in the name. That version is derived from the git metadata and, as long as you are in the git tree you have it in the package too.

If you create an sdist that does not ship enough git metadata then you don't get a correct version. Some folks use the .git_archive.txt (or something similar) but I prefer the write to you that you implemented here.

TL;DR Ideally we need setuptools_scm in the pyproject.toml too to cover all cases.

Reply:
This is fixed now. I moved the entire thing to pyproject.toml